data/json/encounters/randenc_caravans.json

```diff
@@ -0,0 +1,69 @@
+[
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_RandEnc_Roadstop_add",
+    "global": true,
+    "eoc_type": "OM_MOVE",
+    "condition": {
+      "and": [
+        { "u_near_om_location": "roadstop_a", "range": 2 },
+        "is_day",
+        { "one_in_chance": 10 },
+        { "days_since_cataclysm": 7 },
+        {
+          "not": { "u_compare_time_since_var": "RandEnc", "type": "timer", "context": "caravan", "op": "<", "time": "1 d" }
+        },
+        { "not": { "is_season": "winter" } }
+      ]
+    },
+    "effect": [
+      { "mapgen_update": "nest_RandEnc_roadstop_a_add", "om_terrain": "roadstop_a" },
+      { "u_add_var": "RandEnc", "type": "timer", "context": "caravan", "time": true }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_RandEnc_Roadstop_remove",
+    "recurrence": 200,
+    "global": true,
+    "condition": {
+      "and": [ { "u_compare_time_since_var": "RandEnc", "type": "timer", "context": "caravan", "op": ">", "time": "1 h" } ]
+    },
+    "effect": [
+      { "mapgen_update": "nest_RandEnc_roadstop_a_remove", "om_terrain": "roadstop_a" },
+      { "remove_npc": "FM_caravan_merchant_random" },
+      { "remove_npc": "FM_caravan_guard_A" },
+      { "remove_npc": "FM_caravan_guard_B" }
+    ]
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "update_mapgen_id": "nest_RandEnc_roadstop_a_add",
+    "//": "Adds a random encounter at this roadstop",
+    "object": {
+      "place_npcs": [
+        { "class": "FM_caravan_merchant_random", "x": 5, "y": 4 },
+        { "class": "FM_caravan_guard_A", "x": 4, "y": 1 },
+        { "class": "FM_caravan_guard_B", "x": 4, "y": 2 }
+      ],
+      "place_vehicles": [
+        {
+          "vehicle": "pickup",
+          "chance": 100,
+          "fuel": 50,
+          "faction": "free_merchants",
+          "x": [ 4 ],
+          "y": [ 1 ]
+        }
+      ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "update_mapgen_id": "nest_RandEnc_roadstop_a_remove",
+    "//": "removes a random encounter at this roadstop",
+    "object": { "remove_vehicles": [ { "vehicles": [ "pickup" ], "x": [ 1, 15 ], "y": [ 1, 10 ] } ] }
```

Comment:
Could you please explain why `place_vehicles` has strict `x` and `y` coordinates, but `remove_vehicles` has an array of coordinates?

Reply:
No strong reason, but it allows different shaped encounters to go in this area. I don't plan on having them all look identical each time.

Reply:
I mean, according to the code, pickup will be placed at one fixed point every encounter, right?

Reply:
For the moment yes, but not forever. Also it's possible for the truck to move during gameplay but still be in a situation where we want it to despawn, eg. if the player nudges it with another vehicle.